state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -563,15 +536,21 @@ pub mod pallet {
 	/// Determines the action to take when a deposit is made to a channel.
 	#[allow(clippy::large_enum_variant)]
 	#[derive(Clone, RuntimeDebug, PartialEq, Eq, Encode, Decode, TypeInfo)]
+	#[n_functor::derive_n_functor(CcmMetadata = ccm_metadata)]
 	#[scale_info(skip_type_params(C))]
 	#[allow(clippy::large_enum_variant)]
-	pub enum ChannelAction<AccountId, C: Chain> {
+	pub enum ChannelAction<AccountId, ChainAccount, CcmMetadata = CcmChannelMetadataChecked> {
 		Swap {
 			destination_asset: Asset,
 			destination_address: ForeignChainAddress,
+			#[map_with(|arg, f| map_bounded_vec(arg, |x: Beneficiary<_>| x.map(f)))]
 			broker_fees: Beneficiaries<AccountId>,
-			channel_metadata: Option<CcmChannelMetadataChecked>,
-			refund_params: ChannelRefundParametersCheckedInternal<AccountId>,
+			channel_metadata: Option<CcmMetadata>,
+			#[map_with(|arg: ChannelRefundParameters<_,_>, f0, f1| arg.map(|x: AccountOrAddress<_,_>| x.map(f0, |a| a), |z: Option<_>| z.map(f1)))]
```

Comment:
Personally I find this less readable than the more explicit version he had before 🤷

Reply:
Yeah you're totally right about this line, I'll try to make it more readable.